tests/product/image_builder.py

```diff
@@ -0,0 +1,171 @@
+            #
+            # We do this to save the cost of running the installers on the
+            # docker containers every time we run a test. In practice,
+            # that turns out to be a fairly expensive thing to do.
+            if not bare_cluster:
+                self._apply_post_install_hooks(installers)
+                self._update_replacement_keywords(installers)
+                return self.caller.cluster
+
+        # If we got a bare cluster back, we need to run the installers on it.
+        # applying the post-install hooks and updating the replacement
+        # keywords is handled internally in _run_installers.
+        self._run_installers(installers)
+
+        if isinstance(self.caller.cluster, DockerCluster):
+            self.caller.cluster.commit_images(bare_image_provider, cluster_type)
+
+        return self.caller.cluster
+
+    def _setup_cluster_with_no_hadoop_provider(self, cluster_type):
+        return self._setup_cluster(NoHadoopBareImageProvider(),
+                                   cluster_type)
+
+    def setup_standalone_presto_images(self):
+        cluster_type = self.STANDALONE_PRESTO_CLUSTER
+        return self._setup_cluster_with_no_hadoop_provider(cluster_type)
+
+    def setup_standalone_pa_only_images(self):
+        cluster_type = self.PA_ONLY_CLUSTER
+        return self._setup_cluster_with_no_hadoop_provider(cluster_type)
+
+    def setup_standalone_bare_images(self):
+        cluster_type = self.BARE_CLUSTER
+        return self._setup_cluster_with_no_hadoop_provider(cluster_type)
+
+    def setup_yarn_slider_pa_only_images(self):
+        cluster_type = self.PA_ONLY_YS_CLUSTER
+        return self._setup_cluster_with_no_hadoop_provider(cluster_type)
+
+    def _setup_cluster_with_hdp_image_provider(self, cluster_type):
+        return self._setup_cluster(HdpBareImageProvider(),
+                                   cluster_type)
+
+    def setup_yarn_slider_pa_slider_images(self):
+        cluster_type = self.PA_SLIDER_CLUSTER
+        return self._setup_cluster_with_hdp_image_provider(cluster_type)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("image_type", metavar="image_type", type=str, nargs="+",
+                        choices=["standalone_presto", "standalone_pa", "standalone_bare",
+                                 "yarn_slider_pa", "yarn_slider_pa_slider", "all"],
+                        help="Specify the type of image to create. The available choices are: "
+                             "standalone_presto, standalone_pa, standalone_bare, "
+                             "yarn_slider_pa, yarn_slider_pa_slider, all")
+
+    # ImageBuilder needs an input testcase with access to unittest assertions
+    # so the installers can check their resulting installations as well as some
+    # product test helper functions.
+    # This supplies a dummy testcase
+    dummy_testcase = BaseProductTestCase('__init__')
```

Comment:
It's the dummy product test case since the installers need one to invoke assertions like assertIn() and some of the product test helper functions like upload_topology()